chain/client/src/stateless_validation/partial_witness/partial_witness_actor.rs

```diff
@@ -425,7 +455,20 @@ impl PartialWitnessActor {
                     return Ok(());
                 }
             };
-            // TODO(#11099): precompile contracts using self.runtime
+            let runtime = self.runtime.clone();
+            self.compile_contracts_spawner.spawn("precompile_deployed_contracts", move || {
+                match runtime.precompile_contracts(&key.epoch_id, contract_codes) {
```

Comment:
Can you briefly explain the advantage of executing `precompile_contracts` in the async spawner?

Reply:
`precompile_contracts` runs each compilation in a parallel thread, but waits for all threads to finish, so there is a join point before `precompile_contracts` returns. But we do not need to wait for compilations to end (that is lower priority compared to distributing and processing witnesses), and also not block the actor for other tasks. Thus, spawning that entire computation in a separate thread.

or are you asking between different types of spawners?

Reply:
Thanks, I understand now!